libs/http-ext/src/lib/matchers/matcher.ts

```diff
@@ -0,0 +1,8 @@
+export interface MatcherArgs {
+  [key: string]: any;
+}
+
+export interface Matcher<TMatchExpression = any> {
+  canHandle(matchExpression: TMatchExpression): boolean;
+  handle(args: { matchExpression: TMatchExpression } & MatcherArgs): boolean;
```

Comment:
Not sure about the types here.  :thinking: 

```ts
export interface MatcherArgs {
  [key: string]: any;
}
```

Maybe it's too permissive, what do you think @yjaaidi ?

Reply:
I changed the signature, now the handler take two generic arguments:
```ts
handle(args: { expression: TExpression; matchExpression: TMatchExpression; })
``` 
It's to avoid passing different object shape as argument to each matcher.

Reply:
Perfect! 👌 I would just rename `expression` and `TExpression` to `value` and `TValue` as `matchExpression` is the expression we are trying to match to with the given values.
What do you think?